python/ray/dag/dag_node.py

```diff
@@ -58,12 +60,41 @@ def __init__(
         self._bound_other_args_to_resolve: Optional[Dict[str, Any]] = (
             other_args_to_resolve or {}
         )
+
+        # The list of nodes that use this DAG node as input.
+        self._downstream_nodes: List["DAGNode"] = []
+        # The list of nodes that this DAG node uses as input.
+        self._upstream_nodes: List["DAGNode"] = self._prepare_upstream_nodes()
+
         # UUID that is not changed over copies of this node.
         self._stable_uuid = uuid.uuid4().hex
         # Cached values from last call to execute()
         self.cache_from_last_execute = {}
 
         self._type_hint: Optional[ChannelOutputType] = ChannelOutputType()
+        self.is_output_node = False
```

Comment:
Add a comment to be clear output refers to DAG output

Reply:
update: https://github.com/ray-project/ray/pull/47261/commits/cfeb6020eb5951916ce4ee9a8db9a20a1cdd04ae